vhdbuilder/packer/buildperformance/pkg/service/helper.go

```diff
@@ -0,0 +1,198 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+// Set conditions for program to run successfully wtih SetupConfig and CreateDataMaps functions
+func SetupConfig() (*Config, error) {
+	envVars := map[string]string{
+		"kustoTable":                os.Getenv("BUILD_PERFORMANCE_TABLE_NAME"),
+		"kustoEndpoint":             os.Getenv("BUILD_PERFORMANCE_KUSTO_ENDPOINT"),
+		"kustoDatabase":             os.Getenv("BUILD_PERFORMANCE_DATABASE_NAME"),
+		"kustoClientId":             os.Getenv("BUILD_PERFORMANCE_CLIENT_ID"),
+		"kustoIngestionMapping":     os.Getenv("BUILD_PERFORMANCE_INGESTION_MAPPING"),
+		"sourceBranch":              os.Getenv("GIT_BRANCH"),
+		"sigImageName":              os.Getenv("SIG_IMAGE_NAME"),
+		"localBuildPerformanceFile": os.Getenv("SIG_IMAGE_NAME") + "-build-performance.json",
+	}
+	missingVar := false
+	for name, value := range envVars {
+		if value == "" {
+			fmt.Printf("missing environment variable \"%s\".", name)
+			missingVar = true
+		}
+	}
+	if missingVar {
+		return nil, fmt.Errorf("required environment variables were not set")
+	}
+
+	return &Config{
+		KustoTable:                envVars["kustoTable"],
+		KustoEndpoint:             envVars["kustoEndpoint"],
+		KustoDatabase:             envVars["kustoDatabase"],
+		KustoClientId:             envVars["kustoClientId"],
+		KustoIngestionMapping:     envVars["kustoIngestionMapping"],
+		SigImageName:              envVars["sigImageName"],
+		LocalBuildPerformanceFile: envVars["localBuildPerformanceFile"],
+		SourceBranch:              envVars["sourceBranch"],
+	}, nil
+}
+
+func CreateDataMaps() *DataMaps {
+	return &DataMaps{
+		// LocalPerformanceDataMap will hold the performance data from the local JSON file
+		LocalPerformanceDataMap: map[string]map[string]float64{},
+		// QueriedPerformanceDataMap will hold the aggregated performance data queried from Kusto
+		QueriedPerformanceDataMap: map[string]map[string][]float64{},
+		// RegressionMap will hold all identified regressions in the current build
+		RegressionMap: map[string]map[string]float64{},
+	}
+}
+
+// Prepare performance data for evaluation with PreparePerformanceDataForEvaluation and associated helper functions
+func (maps *DataMaps) PreparePerformanceDataForEvaluation(localBuildPerformanceFile string, queriedData *SKU) error {
+	err := maps.DecodeLocalPerformanceData(localBuildPerformanceFile)
```

Comment:
nit: in most cases where the func is only returning an error, I like doing something like this instead:

```
if err := maps.DecodeLocalPerformanceData(localBuildPerformanceFile); err != nil {
    ...
}
```

the `err` will then only be scoped to the if-block

Reply:
Implemented this suggestion throughout the program, though it did make the `IngestData` call in `main.go` a bit long. Its definition is also long. Will stack arguments/parameters if you think its appropriate. 

I also considered just passing in the whole config object to make the function calls/signatures a bit shorter, but I understand we generally aren't supposed to expose fields to a function if the function doesn't need it. I briefly considered making every field a constant to prevent issues, but ended up just leaving it as is for review. Any preference? @cameronmeissner